README.md

```diff
@@ -646,8 +653,99 @@ the user can define endpoints via a
+resource). Then Rancher runs a middleware handler that routes incoming requests
 to defined endpoints. The external services follow the same process of using a
 defined secret containing a URL and token to connect and authenticate to
 Rancher. This aggregation is defined independently and does not use steve's
 aggregation client.
+
+### Design of List Processing API
+
+Steve supports query parameters `filter`, `sort`, `page`/`pagesize`/`revision`,
+and `projectsornamespaces` for list requests as described
+[above](#query-parameters). These formatting options exist to allow user
+interfaces like dashboards to easily consume and display list data in a
+friendly way.
+
+This feature relies on the concept of [stores](#stores) and the RBAC
+partitioner. The [proxy
+store](https://pkg.go.dev/github.com/rancher/steve/pkg/stores/proxy#Store)
+provides raw access to Kubernetes and returns data as an
+[unstructured.UnstructuredList](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured#UnstructuredList).
+The
+[partitioner](https://pkg.go.dev/github.com/rancher/steve/pkg/stores/partition#Partitioner)
+calls the
+proxy store in parallel for each segment of resources the user has access to,
+such as for each namespace. The partitioner feeds the results of each parallelized
+request into a stream of
+[unstructured.Unstructured](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured#Unstructured).
+From here, the list is passed to the
+[listprocessor](https://pkg.go.dev/github.com/rancher/steve/pkg/stores/partition/listprocessor)
+to filter, sort, and paginate the list. The partition store formats the list as
+a
+[types.APIObjectList](https://pkg.go.dev/github.com/rancher/apiserver/pkg/types#APIObjectList)
+and it is returned up the chain of nested stores.
+
+Most stores in steve are implementations of the apiserver
+[Store](https://pkg.go.dev/github.com/rancher/apiserver/pkg/types#Store)
+interface, which returns apiserver
+[types](https://pkg.go.dev/github.com/rancher/apiserver/pkg/types). The
+partitioner implements its own store type called
+[UnstructuredStore](https://pkg.go.dev/github.com/rancher/steve/pkg/stores/partition#UnstructuredStore)
+which returns
+[unstructured.Unstructured](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured#Unstructured)
+objects. The reason for this is that the filtering and sorting functions in the
+listprocessor package need to operate on unstructured data because they work on
+arbitrary fields. However, it also needs to be run after the parallelized
+partitioner has accumulated all the results, because each concurrent fetcher
+will only contain partial results. Therefore, the data remains in an
+unstructured format until after the listprocessor has been run, then the data
+is converted to a structured type. The below diagram illustrates the conversion
+sequence.
+
+![](./docs/store-flow.svg)
+
+#### Unit tests
```

Comment:
Nit: While I think that the section on integration tests is useful (since those are in a separate repo in a hard-to-identify location), I'm not sure that I feel the same way on the unit tests. As long as you know what packages the code is in, it's easy to identify the test files. I would recommend just identifying the packages that are relevant to the list processor instead of including the section on unit tests.

Reply:
It may be obvious where the unit tests are, but I think it's less obvious what to put in those unit tests. In some cases making changes to `pkg/stores/partition/listprocessor` could have an effect on the output of `pkg/stores/partition/store.go`, and it's worthwhile to add tests in both places even if the code in store.go didn't change (even if that's stretching the definition of "unit"), so I think it's valuable to keep that recommendation here.